File: 101/js/test/fibonacci.test.js
const expect = require('chai').expect;
const fibonacci = require('../fibonacci');

describe('Fibonacci', () => {
  it('should calculate fibonacci number for index in sequence', () => {
    expect(fibonacci(0)).to.be.equal(1);
    expect(fibonacci(1)).to.be.equal(1);
    expect(fibonacci(2)).to.be.equal(2);
    expect(fibonacci(3)).to.be.equal(3);
    expect(fibonacci(4)).to.be.equal(5);
    expect(fibonacci(5)).to.be.equal(8);
  });
});

File: 101/scala/src/test/scala/FibonacciTest.scala
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class  FibonacciTest extends AnyFunSuite with Matchers {
  test("Computes fibonacci number") {
    Fibonacci(0) should be(0)
    Fibonacci(1) should be(1)
    Fibonacci(2) should be(1)
    Fibonacci(3) should be(2)
    Fibonacci(4) should be(3)
    Fibonacci(5) should be(5)
    Fibonacci(6) should be(8)
  }
}
